presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergSplitManager.java

```diff
@@ -0,0 +1,249 @@
+
+        TransactionManager transactionManager = getQueryRunner().getTransactionManager();
+        SplitManager splitManager = getQueryRunner().getSplitManager();
+
+        // Get a plan for query on <tableName> with filter `a = date '1984-03-01'`
+        // There should leave only 1 split after file scanning with the filter
+        validateSplitsPlannedForSql(splitManager, transactionManager, filterPushdown,
+                "select * from " + tableName + " where a = date '1984-03-01'",
+                1);
+
+        // Get a plan for query on <tableName> with filter `a = date '1984-06-09'`
+        // There should be 2 splits after file scanning with the filter
+        validateSplitsPlannedForSql(splitManager, transactionManager, filterPushdown,
+                "select * from " + tableName + " where a = date '1984-06-09'",
+                2);
+
+        // Get a plan for query on <tableName> with filter `a = date '1984-06-09' and b = 8`
+        // There should leave only 1 split after file scanning with the filter
+        validateSplitsPlannedForSql(splitManager, transactionManager, filterPushdown,
+                "select * from " + tableName + " where a = date '1984-06-09' and b = 8",
+                1);
+
+        assertQuerySucceeds("DROP TABLE " + tableName);
+    }
+
+    private Session withPushdownFilter(boolean pushdown)
+    {
+        return Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalogSessionProperty(ICEBERG_CATALOG, PUSHDOWN_FILTER_ENABLED, pushdown ? "true" : "false")
+                .build();
+    }
+
+    private void validateSplitsPlannedForSql(SplitManager splitManager,
+                                            TransactionManager transactionManager,
+                                            boolean filterPushdown,
+                                            String sql,
+                                            int expectedSplitCount)
+    {
+        Session session = withPushdownFilter(filterPushdown);
+        List<TableScanNode> tableScanNodes = getTableScanFromOptimizedPlanOfSql(sql, session);
+        assertTrue(tableScanNodes != null && tableScanNodes.size() == 1);
+
+        TransactionId transactionId = transactionManager.beginTransaction(false);
+        session = session.beginTransactionId(transactionId, transactionManager, new AllowAllAccessControl());
+        TableHandle tableHandle = tableScanNodes.get(0).getTable();
+        TableHandle newTableHandle = new TableHandle(tableHandle.getConnectorId(),
+                tableHandle.getConnectorHandle(),
+                transactionManager.getConnectorTransaction(transactionId, tableHandle.getConnectorId()),
+                tableHandle.getLayout(),
+                tableHandle.getDynamicFilter());
+        SplitSource splitSource = splitManager.getSplits(session, newTableHandle, SplitSchedulingStrategy.UNGROUPED_SCHEDULING, WarningCollector.NOOP);
```

Comment:
nit: we can probably replace this by try-with-resources to automatically close splitSource

```suggestion
        try (SplitSource splitSource = splitManager.getSplits(session, newTableHandle, SplitSchedulingStrategy.UNGROUPED_SCHEDULING, WarningCollector.NOOP)) {
        }
        catch (Exception e) {
        }
        finally {
        }
```